File: src/api/src/data/knowledge-base.ts
import type { Scenario, SampleEstimate, ReferenceArchitecture } from '../models/index.js';

export const scenarios: Scenario[] = [
  {
    id: 'SCN-001',
    title: 'Digital Commerce Platform',
    industry: 'Retail',
    description:
      'End-to-end digital commerce solution enabling omnichannel retail experiences with unified inventory, personalized recommendations, and seamless checkout across web, mobile, and in-store.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/guide/architecture-styles/web-queue-worker',
    tags: ['e-commerce', 'omnichannel', 'digital', 'storefront', 'retail', 'shopping', 'POS', 'inventory'],
  },
  {
    id: 'SCN-002',
    title: 'Digital Transformation using AI',
    industry: 'Cross-Industry',
    description:
      'Modernize legacy systems and business processes with AI-powered automation, intelligent document processing, and predictive analytics across any industry vertical.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/ai-ml/guide/data-science-and-machine-learning',
    tags: ['AI', 'machine learning', 'modernization', 'digital', 'transformation', 'automation'],
  },
  {
    id: 'SCN-003',
    title: 'Patient Portal & Telehealth',
    industry: 'Healthcare',
    description:
      'Secure patient engagement platform with telehealth video visits, appointment scheduling, medical records access, and remote monitoring integration for hospitals and clinics.',
    link: 'https://learn.microsoft.com/en-us/industry/healthcare/architecture/patient-access',
    tags: ['patient', 'telehealth', 'hospital', 'clinical', 'EHR', 'medical', 'HIPAA', 'health'],
  },
  {
    id: 'SCN-004',
    title: 'Remote Patient Monitoring',
    industry: 'Healthcare',
    description:
      'IoT-based remote patient monitoring platform that collects vitals from wearable devices, enables telehealth consultations, and provides real-time alerts to care teams.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/solution-ideas/articles/telehealth-system',
    tags: ['IoT', 'telehealth', 'patient', 'monitoring', 'hospital', 'medical', 'wearable', 'health'],
  },
  {
    id: 'SCN-005',
    title: 'Smart Factory IoT Platform',
    industry: 'Manufacturing',
    description:
      'Connected factory solution with IoT sensors, digital twins, predictive maintenance, and real-time production monitoring for Industry 4.0 manufacturing environments.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/guide/iiot-guidance/iiot-architecture',
    tags: ['factory', 'IoT', 'predictive maintenance', 'production', 'supply chain', 'OT', 'manufacturing', 'assembly'],
  },
  {
    id: 'SCN-006',
    title: 'Fraud Detection & Prevention',
    industry: 'Financial Services',
    description:
      'Real-time fraud detection system using machine learning models, anomaly detection, and transaction scoring to protect financial institutions from fraudulent activities.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/ai-ml/architecture/real-time-fraud-detection',
    tags: ['fraud', 'bank', 'fintech', 'real-time analytics', 'anomaly detection', 'security', 'compliance', 'financial'],
  },
  {
    id: 'SCN-007',
    title: 'Citizen Services Portal',
    industry: 'Public Sector',
    description:
      'Unified digital government platform enabling citizens to access services online, submit applications, track requests, and interact with government agencies securely.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/guide/data/data-lake-warehouse',
    tags: ['government', 'citizen', 'public sector', 'digital services', 'portal', 'compliance'],
  },
  {
    id: 'SCN-008',
    title: 'Data Analytics & BI Platform',
    industry: 'Cross-Industry',
    description:
      'Enterprise-grade analytics and business intelligence platform with data warehousing, interactive dashboards, and self-service reporting powered by Azure Synapse and Power BI.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/solution-ideas/articles/azure-databricks-modern-analytics-architecture',
    tags: ['analytics', 'BI', 'data', 'reporting', 'dashboard', 'insights', 'data warehouse'],
  },
  {
    id: 'SCN-009',
    title: 'Intelligent Claims Processing',
    industry: 'Financial Services',
    description:
      'AI-powered claims automation pipeline for insurance companies that extracts data from documents, validates claims, and accelerates processing with minimal human intervention.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/ai-ml/architecture/automate-document-classification-durable-functions',
    tags: ['insurance', 'claims', 'automation', 'AI', 'fintech', 'compliance', 'financial'],
  },
  {
    id: 'SCN-010',
    title: 'Predictive Maintenance Platform',
    industry: 'Manufacturing',
    description:
      'IoT and ML-based predictive maintenance solution that analyzes sensor data to forecast equipment failures and optimize maintenance schedules in manufacturing plants.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/ai-ml/idea/predictive-maintenance',
    tags: ['IoT', 'predictive maintenance', 'factory', 'production', 'OT', 'manufacturing', 'analytics'],
  },
];

export const sampleEstimates: SampleEstimate[] = [
  {
    id: 'EST-001',
    title: 'Calgary Connected Commerce',
    customerName: 'Calgary Co-op',
    industry: 'Retail',
    description:
      'Omnichannel commerce platform connecting 400+ retail locations with unified inventory, mobile ordering, and curbside pickup powered by Azure App Service and Cosmos DB.',
    link: undefined,
    estimatedACR: 420000,
  },
  {
    id: 'EST-002',
    title: 'Costco Phase 2 Digital Transform',
    customerName: 'Costco',
    industry: 'Retail',
    description:
      'Phase 2 of digital transformation initiative migrating legacy warehouse management and e-commerce systems to Azure microservices architecture with AKS and Azure SQL.',
    link: undefined,
    estimatedACR: 1200000,
  },
  {
    id: 'EST-003',
    title: 'New Balance Project Dawn',
    customerName: 'New Balance',
    industry: 'Retail',
    description:
      'Direct-to-consumer digital commerce platform with personalized shopping experiences, AI-driven recommendations, and global CDN deployment on Azure.',
    link: undefined,
    estimatedACR: 680000,
  },
  {
    id: 'EST-004',
    title: 'HSBC Fraud Detection System',
    customerName: 'HSBC',
    industry: 'Financial Services',
    description:
      'Real-time fraud detection and prevention system processing millions of transactions daily using Azure Databricks, Event Hubs, and custom ML models.',
    link: undefined,
    estimatedACR: 890000,
  },
  {
    id: 'EST-005',
    title: 'Cleveland Clinic Telehealth',
    customerName: 'Cleveland Clinic',
    industry: 'Healthcare',
    description:
      'Enterprise telehealth platform enabling virtual visits, remote patient monitoring, and secure health data exchange with HIPAA-compliant Azure infrastructure.',
    link: undefined,
    estimatedACR: 540000,
  },
  {
    id: 'EST-006',
    title: 'Siemens Smart Factory',
    customerName: 'Siemens',
    industry: 'Manufacturing',
    description:
      'Industry 4.0 smart factory deployment with IoT Edge devices, Azure Digital Twins, and predictive maintenance analytics across 12 manufacturing plants.',
    link: undefined,
    estimatedACR: 1500000,
  },
  {
    id: 'EST-007',
    title: 'NYC Digital Services',
    customerName: 'NYC Government',
    industry: 'Public Sector',
    description:
      'Citizen-facing digital services portal consolidating 50+ city services with secure identity verification, case management, and Azure Government cloud deployment.',
    link: undefined,
    estimatedACR: 720000,
  },
  {
    id: 'EST-008',
    title: 'Contoso Insurance Claims AI',
    customerName: 'Contoso Insurance',
    industry: 'Financial Services',
    description:
      'AI-powered claims processing pipeline reducing claim cycle time by 60% using Azure AI Document Intelligence, Logic Apps, and Azure SQL.',
    link: undefined,
    estimatedACR: 350000,
  },
  {
    id: 'EST-009',
    title: 'Fabrikam Health Monitoring',
    customerName: 'Fabrikam Health',
    industry: 'Healthcare',
    description:
      'Remote patient health monitoring system using IoT wearables and Azure IoT Hub to track vitals and trigger automated clinical workflows.',
    link: undefined,
    estimatedACR: 290000,
  },
];

export const referenceArchitectures: ReferenceArchitecture[] = [
  {
    id: 'ARCH-001',
    title: 'Microservices-Based E-Commerce Platform',
    description:
      'Cloud-native e-commerce platform using microservices architecture with AKS for container orchestration, Cosmos DB for product catalog, and API Management for unified API gateway.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/microservices/',
    azureServices: ['Azure Kubernetes Service', 'Cosmos DB', 'API Management', 'Azure Front Door'],
  },
  {
    id: 'ARCH-002',
    title: 'Scalable E-Commerce Web App',
    description:
      'Scalable web application architecture for e-commerce with App Service for hosting, Azure SQL for transactional data, Redis cache for performance, and CDN for global content delivery.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/web-apps/app-service/architectures/baseline-zone-redundant',
    azureServices: ['App Service', 'Azure SQL', 'Azure Cache for Redis', 'CDN'],
  },
  {
    id: 'ARCH-003',
    title: 'AI-Powered Customer Service',
    description:
      'Intelligent customer service solution using Azure AI Document Intelligence and Cognitive Services to automate document processing, with Blob Storage and Logic Apps for workflow orchestration.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/ai-ml/architecture/conversational-bot',
    azureServices: ['Azure AI Document Intelligence', 'Cognitive Services', 'Blob Storage', 'Logic Apps'],
  },
  {
    id: 'ARCH-004',
    title: 'IoT Reference Architecture',
    description:
      'Comprehensive IoT solution architecture with IoT Hub for device management, Stream Analytics for real-time processing, Digital Twins for spatial modeling, and Time Series Insights for historical analytics.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/reference-architectures/iot',
    azureServices: ['IoT Hub', 'Stream Analytics', 'Azure Digital Twins', 'Time Series Insights'],
  },
  {
    id: 'ARCH-005',
    title: 'Healthcare Data Platform',
    description:
      'HIPAA-compliant healthcare data platform for aggregating clinical data, enabling analytics, and supporting telehealth workloads with Azure API for FHIR, Azure Synapse, and Power BI.',
    link: 'https://learn.microsoft.com/en-us/industry/healthcare/architecture/healthcare-data',
    azureServices: ['Azure API for FHIR', 'Azure Synapse Analytics', 'Power BI', 'Azure Data Lake Storage'],
  },
  {
    id: 'ARCH-006',
    title: 'Financial Services Analytics',
    description:
      'Real-time and batch analytics architecture for financial services with Event Hubs for transaction ingestion, Databricks for ML, Synapse for warehousing, and Power BI for reporting.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/industries/finance/',
    azureServices: ['Event Hubs', 'Azure Databricks', 'Azure Synapse Analytics', 'Power BI'],
  },
  {
    id: 'ARCH-007',
    title: 'Government Cloud Architecture',
    description:
      'Secure government cloud architecture designed for FedRAMP and IL5 compliance with Azure Government, Azure Active Directory, Key Vault, and Azure Monitor for comprehensive governance.',
    link: 'https://learn.microsoft.com/en-us/azure/azure-government/documentation-government-overview-azure',
    azureServices: ['Azure Government', 'Azure Active Directory', 'Key Vault', 'Azure Monitor'],
  },
  {
    id: 'ARCH-008',
    title: 'Real-Time Fraud Detection',
    description:
      'Real-time fraud detection architecture processing streaming transactions with Event Hubs, running ML scoring via Azure Databricks, and storing results in Azure Synapse for investigation.',
    link: 'https://learn.microsoft.com/en-us/azure/architecture/ai-ml/architecture/real-time-fraud-detection',
    azureServices: ['Event Hubs', 'Azure Databricks', 'Azure Synapse Analytics', 'Power BI'],
  },
];

File: src/python-api/data/knowledge_base.py
"""Local reference-architecture patterns conforming to FRD-02 §3.3 schema.

Each pattern carries the full retrieved-pattern schema so it can be used as a
seamless fallback when the MCP-based KnowledgeAgent is unavailable.
"""

PATTERNS: list[dict] = [
    # 1 ── Scalable E-Commerce Web App (Retail)
    {
        "title": "Scalable E-Commerce Web App",
        "url": "https://learn.microsoft.com/azure/architecture/example-scenario/apps/ecommerce-scenario",
        "summary": (
            "A highly scalable e-commerce platform built on Azure App Service "
            "with Azure SQL and Redis Cache for sub-millisecond response times. "
            "Azure CDN delivers static assets globally for fast page loads."
        ),
        "workload_type": "web-app",
        "industry": "Retail",
        "compliance_tags": ["PCI-DSS", "GDPR"],
        "recommended_services": [
            "Azure App Service",
            "Azure SQL Database",
            "Azure Cache for Redis",
            "Azure CDN",
            "Azure Blob Storage",
        ],
        "components": [
            {"name": "Web Frontend", "azureService": "Azure App Service", "description": "Hosts the customer-facing storefront"},
            {"name": "Product Database", "azureService": "Azure SQL Database", "description": "Stores product catalog, orders, and customer data"},
            {"name": "Session Cache", "azureService": "Azure Cache for Redis", "description": "Caches shopping-cart and session state for low latency"},
            {"name": "Static Assets", "azureService": "Azure CDN", "description": "Delivers images, scripts, and stylesheets globally"},
            {"name": "Media Storage", "azureService": "Azure Blob Storage", "description": "Stores product images and uploaded media"},
        ],
        "confidence_score": 1.0,
    },
    # 2 ── AI-Powered Product Recommendations (Retail)
    {
        "title": "AI-Powered Product Recommendations",
        "url": "https://learn.microsoft.com/azure/architecture/ai-ml/idea/product-recommendations",
        "summary": (
            "Uses Azure Machine Learning and Cosmos DB to deliver real-time, "
            "personalized product recommendations. The model is trained on "
            "historical purchase data and served via a managed online endpoint."
        ),
        "workload_type": "ai-ml",
        "industry": "Retail",
        "compliance_tags": ["GDPR"],
        "recommended_services": [
            "Azure Machine Learning",
            "Azure Cosmos DB",
            "Azure Kubernetes Service",
            "Azure Event Hubs",
            "Azure Blob Storage",
        ],
        "components": [
            {"name": "Recommendation Model", "azureService": "Azure Machine Learning", "description": "Trains and serves the collaborative-filtering model"},
            {"name": "User Profile Store", "azureService": "Azure Cosmos DB", "description": "Stores user behavior and preference vectors"},
            {"name": "Serving Infrastructure", "azureService": "Azure Kubernetes Service", "description": "Hosts the real-time inference API"},
            {"name": "Event Ingestion", "azureService": "Azure Event Hubs", "description": "Streams clickstream events for near-real-time retraining"},
        ],
        "confidence_score": 1.0,
    },
    # 3 ── Patient Portal / Telehealth (Healthcare)
    {
        "title": "Patient Portal and Telehealth Platform",
        "url": "https://learn.microsoft.com/azure/architecture/example-scenario/digital-health/health-portal",
        "summary": (
            "A HIPAA-compliant patient portal enabling virtual visits, secure "
            "messaging, and medical-record access. Azure API for FHIR provides "
            "standards-based health-data interoperability."
        ),
        "workload_type": "web-app",
        "industry": "Healthcare",
        "compliance_tags": ["HIPAA", "SOC2"],
        "recommended_services": [
            "Azure App Service",
            "Azure API for FHIR",
            "Azure Active Directory B2C",
            "Azure Cosmos DB",
            "Azure Communication Services",
        ],
        "components": [
            {"name": "Patient Web App", "azureService": "Azure App Service", "description": "Hosts the patient-facing portal with appointment scheduling"},
            {"name": "Health Data API", "azureService": "Azure API for FHIR", "description": "Provides FHIR-compliant access to electronic health records"},
            {"name": "Identity Provider", "azureService": "Azure Active Directory B2C", "description": "Manages patient identity and MFA enrollment"},
            {"name": "Clinical Data Store", "azureService": "Azure Cosmos DB", "description": "Stores clinical notes, messages, and attachments"},
            {"name": "Video Visits", "azureService": "Azure Communication Services", "description": "Powers real-time video telehealth consultations"},
        ],
        "confidence_score": 1.0,
    },
    # 4 ── IoT Telemetry Platform (Manufacturing)
    {
        "title": "IoT Telemetry Platform",
        "url": "https://learn.microsoft.com/azure/architecture/reference-architectures/iot",
        "summary": (
            "Ingests millions of IoT events per second from factory-floor sensors "
            "using IoT Hub and Stream Analytics. Cosmos DB stores device state "
            "while Power BI surfaces operational dashboards."
        ),
        "workload_type": "iot",
        "industry": "Manufacturing",
        "compliance_tags": ["SOC2"],
        "recommended_services": [
            "Azure IoT Hub",
            "Azure Stream Analytics",
            "Azure Cosmos DB",
            "Azure Functions",
            "Power BI",
        ],
        "components": [
            {"name": "Device Gateway", "azureService": "Azure IoT Hub", "description": "Securely connects and manages millions of IoT devices"},
            {"name": "Stream Processing", "azureService": "Azure Stream Analytics", "description": "Performs real-time aggregation and anomaly detection"},
            {"name": "Device State Store", "azureService": "Azure Cosmos DB", "description": "Stores device twins and time-series telemetry"},
            {"name": "Serverless Actions", "azureService": "Azure Functions", "description": "Triggers alerts and downstream workflows on threshold breaches"},
            {"name": "Operational Dashboard", "azureService": "Power BI", "description": "Visualizes KPIs and telemetry trends for plant operators"},
        ],
        "confidence_score": 1.0,
    },
    # 5 ── Real-Time Fraud Detection (Financial Services)
    {
        "title": "Real-Time Fraud Detection",
        "url": "https://learn.microsoft.com/azure/architecture/example-scenario/ai/fraud-detection",
        "summary": (
            "Applies machine-learning models to streaming transaction data to "
            "flag fraudulent activity in near real time. Azure Event Hubs and "
            "Stream Analytics feed scored events into Cosmos DB for investigation."
        ),
        "workload_type": "ai-ml",
        "industry": "Financial Services",
        "compliance_tags": ["SOC2", "PCI-DSS"],
        "recommended_services": [
            "Azure Machine Learning",
            "Azure Event Hubs",
            "Azure Stream Analytics",
            "Azure Cosmos DB",
            "Azure Synapse Analytics",
        ],
        "components": [
            {"name": "Fraud Model", "azureService": "Azure Machine Learning", "description": "Trains and serves the fraud-scoring classification model"},
            {"name": "Transaction Ingestion", "azureService": "Azure Event Hubs", "description": "Streams millions of payment transactions per second"},
            {"name": "Real-Time Scoring", "azureService": "Azure Stream Analytics", "description": "Applies the ML model to each transaction with sub-second latency"},
            {"name": "Case Store", "azureService": "Azure Cosmos DB", "description": "Stores flagged transactions for analyst review"},
            {"name": "Historical Analytics", "azureService": "Azure Synapse Analytics", "description": "Enables retrospective analysis and model-retraining pipelines"},
        ],
        "confidence_score": 1.0,
    },
    # 6 ── Microservices on AKS (Cross-Industry)
    {
        "title": "Microservices on AKS",
        "url": "https://learn.microsoft.com/azure/architecture/reference-architectures/containers/aks-microservices/aks-microservices",
        "summary": (
            "Cloud-native microservices architecture on Azure Kubernetes Service "
            "with API Management as the gateway. Supports independent scaling, "
            "CI/CD per service, and centralized observability via Azure Monitor."
        ),
        "workload_type": "microservices",
        "industry": "Cross-Industry",
        "compliance_tags": ["SOC2"],
        "recommended_services": [
            "Azure Kubernetes Service",
            "Azure Container Registry",
            "Azure API Management",
            "Azure Monitor",
            "Azure Key Vault",
        ],
        "components": [
            {"name": "Container Orchestration", "azureService": "Azure Kubernetes Service", "description": "Runs and scales containerized microservices"},
            {"name": "Image Registry", "azureService": "Azure Container Registry", "description": "Stores and scans container images"},
            {"name": "API Gateway", "azureService": "Azure API Management", "description": "Provides rate limiting, auth, and API versioning"},
            {"name": "Observability", "azureService": "Azure Monitor", "description": "Collects logs, metrics, and distributed traces"},
            {"name": "Secrets Management", "azureService": "Azure Key Vault", "description": "Stores certificates, connection strings, and API keys"},
        ],
        "confidence_score": 1.0,
    },
    # 7 ── Modern Data Warehouse (Cross-Industry)
    {
        "title": "Modern Data Warehouse",
        "url": "https://learn.microsoft.com/azure/architecture/solution-ideas/articles/enterprise-data-warehouse",
        "summary": (
            "Enterprise analytics solution combining Synapse Analytics, Data Lake "
            "Storage, and Data Factory for ETL orchestration. Power BI delivers "
            "self-service dashboards to business users."
        ),
        "workload_type": "data-platform",
        "industry": "Cross-Industry",
        "compliance_tags": ["SOC2", "GDPR"],
        "recommended_services": [
            "Azure Synapse Analytics",
            "Azure Data Lake Storage",
            "Azure Data Factory",
            "Power BI",
            "Microsoft Purview",
        ],
        "components": [
            {"name": "Analytics Engine", "azureService": "Azure Synapse Analytics", "description": "Serverless and dedicated SQL pools for large-scale queries"},
            {"name": "Data Lake", "azureService": "Azure Data Lake Storage", "description": "Centralized repository for structured and unstructured data"},
            {"name": "ETL Orchestration", "azureService": "Azure Data Factory", "description": "Manages data ingestion and transformation pipelines"},
            {"name": "Business Intelligence", "azureService": "Power BI", "description": "Interactive dashboards and self-service reporting"},
            {"name": "Data Governance", "azureService": "Microsoft Purview", "description": "Data catalog, lineage tracking, and classification"},
        ],
        "confidence_score": 1.0,
    },
    # 8 ── Multi-Tenant SaaS (Cross-Industry)
    {
        "title": "Multi-Tenant SaaS Application",
        "url": "https://learn.microsoft.com/azure/architecture/guide/multitenant/overview",
        "summary": (
            "A SaaS platform with tenant isolation, per-tenant scaling, and "
            "shared infrastructure on Azure. Azure Front Door provides global "
            "load balancing while Azure AD B2C handles customer identity."
        ),
        "workload_type": "web-app",
        "industry": "Cross-Industry",
        "compliance_tags": ["SOC2", "GDPR"],
        "recommended_services": [
            "Azure App Service",
            "Azure SQL Elastic Pool",
            "Azure Front Door",
            "Azure Key Vault",
            "Azure Active Directory B2C",
        ],
        "components": [
            {"name": "Application Tier", "azureService": "Azure App Service", "description": "Hosts the multi-tenant web API and UI"},
            {"name": "Tenant Databases", "azureService": "Azure SQL Elastic Pool", "description": "Provides per-tenant database isolation with shared resources"},
            {"name": "Global Load Balancer", "azureService": "Azure Front Door", "description": "Routes traffic with WAF protection and SSL offload"},
            {"name": "Secrets Management", "azureService": "Azure Key Vault", "description": "Stores tenant-specific connection strings and secrets"},
            {"name": "Customer Identity", "azureService": "Azure Active Directory B2C", "description": "Manages customer sign-up, sign-in, and profile management"},
        ],
        "confidence_score": 1.0,
    },
    # 9 ── Event-Driven Architecture (Cross-Industry)
    {
        "title": "Event-Driven Architecture",
        "url": "https://learn.microsoft.com/azure/architecture/guide/architecture-styles/event-driven",
        "summary": (
            "Loosely coupled system using Event Grid for pub/sub routing and "
            "Service Bus for reliable message queuing. Azure Functions provide "
            "serverless event handlers that scale automatically."
        ),
        "workload_type": "microservices",
        "industry": "Cross-Industry",
        "compliance_tags": ["SOC2"],
        "recommended_services": [
            "Azure Event Grid",
            "Azure Service Bus",
            "Azure Functions",
            "Azure Cosmos DB",
            "Azure Monitor",
        ],
        "components": [
            {"name": "Event Router", "azureService": "Azure Event Grid", "description": "Routes events to subscribers with filtering and fan-out"},
            {"name": "Message Broker", "azureService": "Azure Service Bus", "description": "Provides reliable queuing with dead-letter and session support"},
            {"name": "Event Handlers", "azureService": "Azure Functions", "description": "Serverless compute that processes events on demand"},
            {"name": "State Store", "azureService": "Azure Cosmos DB", "description": "Persists event-sourced state with multi-region replication"},
            {"name": "Observability", "azureService": "Azure Monitor", "description": "End-to-end distributed tracing across event flows"},
        ],
        "confidence_score": 1.0,
    },
    # 10 ── Cloud Migration – Lift & Shift (Cross-Industry)
    {
        "title": "Cloud Migration (Lift and Shift)",
        "url": "https://learn.microsoft.com/azure/architecture/cloud-adoption-framework/migrate/",
        "summary": (
            "Migrates on-premises workloads to Azure IaaS with minimal "
            "refactoring using Azure Migrate for assessment and Azure Site "
            "Recovery for replication. Ideal first step in a modernization journey."
        ),
        "workload_type": "migration",
        "industry": "Cross-Industry",
        "compliance_tags": ["SOC2", "GDPR"],
        "recommended_services": [
            "Azure Migrate",
            "Azure Site Recovery",
            "Azure Virtual Machines",
            "Azure Virtual Network",
            "Azure Monitor",
        ],
        "components": [
            {"name": "Assessment", "azureService": "Azure Migrate", "description": "Discovers and assesses on-premises servers for cloud readiness"},
            {"name": "Replication", "azureService": "Azure Site Recovery", "description": "Replicates VMs to Azure with minimal downtime cutover"},
            {"name": "Compute", "azureService": "Azure Virtual Machines", "description": "Hosts migrated workloads on IaaS virtual machines"},
            {"name": "Networking", "azureService": "Azure Virtual Network", "description": "Provides isolated network with hybrid connectivity"},
            {"name": "Monitoring", "azureService": "Azure Monitor", "description": "Tracks VM performance and health post-migration"},
        ],
        "confidence_score": 1.0,
    },
]

# Backward-compatible alias
SCENARIOS: list[dict] = PATTERNS


def search_local_patterns(query: str, top_k: int = 5) -> list[dict]:
    """Search local patterns using keyword matching. Fallback for MCP unavailability."""
    query_lower = query.lower()
    scored: list[tuple[int, dict]] = []
    for pattern in PATTERNS:
        score = 0
        for field in ["title", "summary", "industry", "workload_type"]:
            val = pattern.get(field, "").lower()
            for word in query_lower.split():
                if word in val and len(word) > 2:
                    score += 1
        for svc in pattern.get("recommended_services", []):
            if any(w in svc.lower() for w in query_lower.split() if len(w) > 2):
                score += 2
        if score > 0:
            result = {**pattern, "confidence_score": min(score / 10, 1.0)}
            scored.append((score, result))

    scored.sort(key=lambda x: x[0], reverse=True)
    if not scored:
        # No keyword matches — return empty instead of forcing irrelevant patterns
        return []
    return [r for _, r in scored[:top_k]]


def find_matching_scenarios(description: str, top_k: int = 3) -> list[dict]:
    """Return the top-k patterns whose fields match the description.

    Kept for backward compatibility – delegates to ``search_local_patterns``.
    """
    return search_local_patterns(description, top_k=top_k)
